src/ipc/ipc4/helper.c

```diff
@@ -787,6 +787,23 @@ int ipc4_add_comp_dev(struct comp_dev *dev)
 	/* add new component to the list */
 	list_item_append(&icd->list, &ipc->comp_list);
 
+	/* If new component added to the pipeline is not fitted for deep buffering we need to
+	 * inform pipeline it cannot work on long periods.
+	 */
+	if (!dev->deep_buffering) {
+		if (!dev->pipeline) {
+			const uint32_t pipe_id = dev->ipc_config.pipeline_id;
+			struct ipc_comp_dev *ipc_comp =
+				ipc_get_comp_by_ppl_id(ipc,
+						       COMP_TYPE_PIPELINE,
+						       pipe_id);
+
+			dev->pipeline = ipc_comp->pipeline;
+		}
+
+		dev->pipeline->long_period = false;
```

Comment:
I was thinking it would be ncie to have a log entry to catch if topology has an incorrect configuration. But as this is called also in normal case, addign a INF level log is probably not a good idea here. It would be good to have a note/warning if host is asking for a combiation with copier configured with deep buffers for DMA, but the pipeline is not compatible.